decompy/ai-rl/RewardFunction.cpp

```diff
@@ -0,0 +1,205 @@
+#include "llvm/IR/Function.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/Pass.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/IR/InstrTypes.h"
+#include "llvm/Transforms/IPO/PassManagerBuilder.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/IRPrintingPasses.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/IR/CallingConv.h"
+#include "llvm/IR/Verifier.h"
+#include "llvm/IR/IRBuilder.h"
+
+#include <math.h>
+#include <cstdlib>
+#include <iostream>
+
+using namespace llvm;
+
+//clang++-8 -g decompy/ai-rl/RewardFunction.cpp `llvm-config-8 --cxxflags --ldflags --libs core` -lpthread -o RewardFunction
+
+class Reward{
+
+    public:
+    int getReward(Function &fnc1, Function &fnc2);
+    int instructionCountReward(Function &fnc1, Function &fnc2);
+    int identicalInstructionTypeCountReward(Function &fnc1, Function &fnc2);
+
+};
+
+int Reward::getReward(Function &fnc1, Function &fnc2){
+    //fnc1 is the function weare trying to be
+    return Reward::instructionCountReward(fnc1, fnc2) + Reward::identicalInstructionTypeCountReward(fnc1, fnc2);
+}
+
+//Returns the percent difference of number of instractions between the two fnc as a neg value
+int Reward::instructionCountReward(Function &fnc1, Function &fnc2){
+    double f1ins = fnc1.getInstructionCount();
+    double f2ins = fnc2.getInstructionCount();
+
+    //find the percent difference between num of instructions in f1 and f2
+    int percent = round(f2ins/f1ins*100);
+    int perDif = -abs(percent-100);
+
+    return perDif;
+}
+
+//returns the # of instructions in each fnnc with same type. cant get identical to to work tho :(
```

Comment:
Is this comment still relevant? I see identical code at the bottom. I'm not sure if that space between in2-> getOpCode where `if(in2-> getOpcode() == in1->getOpcode()) {` is a problem or not.

Reply:
Yeah, it doesn't check if the instructions are identical, it checks if they are the same type (ie the op code)